source/simulator/melt.cc

```diff
@@ -0,0 +1,1092 @@
+/*
```

Comment:
I guess I'm just going to trust you on this file -- I don't understand 3/4 of it, and don't feel competent to actually review it for correctness.